dmn-engine/src/main/scala/org/camunda/dmn/evaluation/LiteralExpressionEvaluator.scala

```diff
@@ -16,10 +21,14 @@ class LiteralExpressionEvaluator(feelEngine: FeelEngine) {
   def evalExpression(literalExpression: ParsedLiteralExpression,
                      context: EvalContext): Either[Failure, Val] = {
 
-    evalExpression(literalExpression.expression, context).right.map { result =>
-      context.audit(literalExpression, SingleEvaluationResult(result))
-
-      result
+    evalExpression(literalExpression.expression, context) match {
+      case r @ Right(result) =>
+        context.audit(literalExpression, SingleEvaluationResult(result))
+        r
+      case l @ Left(failure) =>
+        context.audit(literalExpression,
+                      SingleEvaluationResult(ValError(failure.message)))
+        l
```

Comment:
With the suggested audit function, this can be simplified to:

``` 
    val result = evalExpression(literalExpression.expression, context)
    
    context.audit(literalExpression, result)
    result
```